lib/matplotlib/axes/_base.py

```diff
@@ -985,14 +1040,30 @@ def clear(self):
         """clear the axes"""
         self.cla()
 
+    def set_prop_cycle(self, prop_cycle):
+        """
+        Set the prop cycle for any future plot commands on this Axes.
+
+        *prop_cycle* is a :class:Cycler object.
+        Can also be `None` to reset to the cycle defined by the
+        current style.
+        """
+        self._get_lines.set_prop_cycle(prop_cycle)
+        self._get_patches_for_fill.set_prop_cycle(prop_cycle)
+
     def set_color_cycle(self, clist):
         """
         Set the color cycle for any future plot commands on this Axes.
 
         *clist* is a list of mpl color specifiers.
+
+        .. deprecated:: 1.5
```

Comment:
I don't think so... I see we have a choice of two options.

1. Deprecate this, albeit over a long timescale...
2. Accept we will keep this as part of the codebase and thus patch it up to make sense of the function name in light of us adding support to to internally cycle other properties. and extend this to other properties.  We can do this through the ``__set__`` method.

I don't see the middle way of keeping a function we consider useless -- given by the fact you said you don't want to extend it to other properties -- around forever, just because a lot of people use it currently without the new function.  A lot of people use ``NavigationToolbar2``, but we will deprecate that after one cycle.